helm/grafana/templates/postgresql-cluster.yaml

```diff
@@ -0,0 +1,122 @@
+{{- if .Values.grafanaDatabase.enabled }}
+# Postgresql cluster
+apiVersion: postgresql.cnpg.io/v1
+kind: Cluster
+metadata:
+  name: {{ .Chart.Name }}-postgresql
+  namespace: {{ .Release.Namespace }}
+spec:
+  instances: 2
+  storage:
+  size: 1Gi
+  backup:
+    barmanObjectStore:
+      data:
+        compression: gzip
+      wal:
+        compression: gzip
+      {{- if (eq .Values.grafanaDatabase.provider "capa") }}
+      destinationPath: s3://giantswarm-{{ .Values.grafanaDatabase.baseUrl }}-{{ .Chart.Name }}-postgresql/
+      s3Credentials:
+        inheritFromIAMRole: true
+      {{- end }}
+      {{- if (eq .Values.grafanaDatabase.provider "capz") }}
+      destinationPath: https://${accountName}.blob.core.windows.net/giantswarm-{{ .Values.grafanaDatabase.baseUrl }}-{{ .Chart.Name }}-postgresql
+      azureCredentials:
+        storageAccount:
+          name: giantswarm-{{ .Values.grafanaDatabase.baseUrl }}-{{ .Chart.Name }}-postgresql
+          key: accountName
+        storageKey:
+          name: giantswarm-{{ .Values.grafanaDatabase.baseUrl }}-{{ .Chart.Name }}-postgresql
+          key: accountKey
+      {{- end }}
+    retentionPolicy: "30d"
+    {{- if (eq .Values.grafanaDatabase.provider "capa") }}
+    serviceAccountTemplate:
+      metadata:
+        annotations:
+          eks.amazonaws.com/role-arn: {{ .Values.grafanaDatabase.irsaPrefix }}:role/giantswarm-{{ .Values.grafanaDatabase.baseUrl }}-{{ .Chart.Name }}-postgresql
```

Comment:
How about we can override the full annotations block?

.Values.grafanaDatabase.serviceAccount.annotations?

Reply:
I would prefer not to because this way we can keep the configurable `Chart.Name` and `postgresql` suffix that will be configurable